cogs/roles.py

```diff
@@ -32,132 +32,89 @@ def __init__(self, bot):
         self.bot = bot
         self.roles = self.bot.config.roles
 
-    @commands.command()
-    async def pronoun(self, ctx, transaction, pronoun):
+    async def toggleRole(self, ctx, transaction, requestedRole, category):
         """
-        Self-assign a pronoun role to a user. 
+        Assigs a single role to a user with no checks from a category of roles
         """
         member = ctx.message.author
+        roles = self.roles.get(category).split(", ")
 
-        # normalize pronoun
-        pronoun = pronoun.title()
-
-        pronouns = self.roles.get("pronouns").split(", ")
-
-        if (transaction == "add") and (pronoun in pronouns):
-            role = discord.utils.get(ctx.guild.roles, name=pronoun)
+        # normalize user input to title case
+        requestedRole = requestedRole.title()
 
+        if (transaction == "add") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
             await member.add_roles(role, reason="Self Requested")
-        elif (transaction == "remove") and (pronoun in pronouns):
-            role = discord.utils.get(ctx.guild.roles, name=pronoun)
+        elif (transaction == "remove") and (requestedRole in roles):
+            role = discord.utils.get(ctx.guild.roles, name=requestedRole)
             await member.remove_roles(role, reason="Self Requested")
-
         else:
```

Comment:
Should we have an `elif` for a different error message sent back if the transaction is invalid (if it's not "add" or "remove")?

Reply:
Do I need a different error for add/remove being wrong vs. the role not existing or is one catch all error sufficient?

Reply:
I was thinking that it could be nice for the user of the command to know if they typed in a non-existant role or if they just used the command wrong. But yeah this is super subjective

Reply:
Going to go with a more generic error message catch all here.